examples/jsm/renderers/webgpu/WebGPUTextures.js

```diff
@@ -288,6 +297,18 @@ class WebGPUTextures {
 		const height = ( image !== undefined ) ? image.height : 1;
 
 		const format = this._convertFormat( texture.type );
+		const needsMipmaps = this._needsMipmaps( texture );
+		const mipLevelCount = ( needsMipmaps === true ) ? this._computeMipLevelCount( width, height ) : undefined;
+
+		let usage = GPUTextureUsage.SAMPLED | GPUTextureUsage.COPY_DST;
+
+		if ( needsMipmaps === true ) {
+
+			usage |= GPUTextureUsage.OUTPUT_ATTACHMENT;
```

Comment:
That's unfortunate because OUTPUT_ATTACHMENT can make the texture consume a whole lot more memory. I think @toji's utility does a copy when the texture isn't OUTPUT_ATTACHMENT so that it doesn't pessimize things.

Reply:
It seems this code is required:

https://github.com/toji/web-texture-tool/blob/b41aa7617d5d3040ff99fd2856940c2205492f86/src/webgpu-texture-tool.js#L124-L127

Reply:
My older code always did the copy, my newer code always does the render between texture layers. The ideal scenario would be to have both paths and choose based on whether OUTPUT_ATTACHMENT was used at creation. It's on my to-do list, at least.

Reply:
The code is definitely a good start for `WebGPUTextureUtils` 😊 .

Reply:
This thread gave me the prodding I needed to make the change anyway. :) https://github.com/toji/web-texture-tool/blob/master/src/webgpu-mipmap-generator.js will now handle mipmapping for texture creation for textures created without OUTPUT_ATTACHMENT usage. It's still not clear to me which path is preferrable, and I suspect it depends on the hardware?

The OUTPUT_ATTACHMENT path is going to be faster to generate mipmaps for, but *may* be slower to access afterward. The copy path will be slower during mipmap gen and use more memory but *may* allow for faster sampling later? Maybe @Kangz has some better insight into the pros/cons?

Reply:
`OUTPUT_ATTACHMENT` allows in-place mipmap generation and avoids the allocation and release of temporary texture memory, but using it when not necessary will consume more memory (due to additional compression planes and stricter alignment requirements) and could prevent some faster HW paths in some cases.